tests/components/samsungtv/test_config_flow.py

```diff
@@ -0,0 +1,388 @@
+
+
+async def test_autodetect_websocket(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch("homeassistant.components.samsungtv.config_flow.Remote") as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "create_entry"
+        assert result["data"][CONF_METHOD] == "websocket"
+        assert remote.call_count == 1
+        assert remote.call_args_list == [call(AUTODETECT_WEBSOCKET)]
+
+
+async def test_autodetect_auth_missing(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=[AccessDenied("Boom")],
+    ) as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "auth_missing"
+        assert remote.call_count == 1
+        assert remote.call_args_list == [call(AUTODETECT_WEBSOCKET)]
+
+
+async def test_autodetect_not_supported(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=[UnhandledResponse("Boom")],
+    ) as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "not_supported"
+        assert remote.call_count == 1
+        assert remote.call_args_list == [call(AUTODETECT_WEBSOCKET)]
+
+
+async def test_autodetect_legacy(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=[OSError("Boom"), mock.DEFAULT],
+    ) as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "create_entry"
+        assert result["data"][CONF_METHOD] == "legacy"
+        assert remote.call_count == 2
+        assert remote.call_args_list == [
+            call(AUTODETECT_WEBSOCKET),
+            call(AUTODETECT_LEGACY),
+        ]
+
+
+async def test_autodetect_none(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=OSError("Boom"),
+    ) as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "not_found"
+        assert remote.call_count == 2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4795320)